File: Python/Mahasiswa.py
# import kelas Sivitas Akademik
from SivitasAkademik import SivitasAkademik

# deklarasi kelas Mahasiswa yang merupakan child dari kelas Sivitas Akademik
class Mahasiswa(SivitasAkademik):

    # atribut2 privat milik kelas Mahasiswa
    __nim = 0
    __kode_prodi = 0

    # constructor
    def __init__(
        self,
        nik=0,
        nama="",
        jenis_kelamin="",
        asal_universitas="",
        email_edu="",
        fakultas="",
        nim=0,
        kode_prodi=0,
    ):
        # constructor super untuk kelas Sivitas Akademik
        super().__init__(
            nik, nama, jenis_kelamin, asal_universitas, email_edu, fakultas
        )
        self.__nim = nim
        self.__kode_prodi = kode_prodi

    # setter nim
    def setNim(self, nim):
        self.__nim = nim

    # setter kode prodi
    def setKodeProdi(self, kode_prodi):
        self.__kode_prodi = kode_prodi

    # getter nim
    def getNim(self):
        return self.__nim

    # getter kode prodi
    def getKodeProdi(self):
        return self.__kode_prodi

File: CPP/Mahasiswa.cpp
#include "header.hh"

/* Definisi Method Kelas Mahasiswa */

// constructor kosong
Mahasiswa::Mahasiswa()
    : SivitasAkademik()
{
}

// constructor dengan parameter seluruh atribut kelas dan super kelas dari Mahasiswa
Mahasiswa::Mahasiswa(int nik, string nama, string jenis_kelamin, string asal_universitas, string email_edu, string fakultas, int nim, int prodi)
    // constructor super dari kelas parent Sivitas Akademik
    : SivitasAkademik(nik, nama, jenis_kelamin, asal_universitas, email_edu, fakultas)
{

    this->nim = nim;
    this->prodi = prodi;
}

// setter nim
void Mahasiswa::setNim(int nim)
{
    this->nim = nim;
}

// setter prodi
void Mahasiswa::setProdi(int prodi)
{
    this->prodi = prodi;
}

// getter nim
int Mahasiswa::getNim()
{
    return this->nim;
}

// getter prodi
int Mahasiswa::getProdi()
{
    return this->prodi;
}

// destructor
Mahasiswa::~Mahasiswa()
{
}
